casper/src/main/resources/MultiSigRevVault.rho

```diff
@@ -0,0 +1,212 @@
+                    contract multiSig(@"transfer", @targetRevAddress, @amount, @authKey, ret) = {
+                      new idxAuthKeysCh, idxAuthPubKeysCh in {
+                        // Check that authKey is in the list of authKeys or authPubKeys
+                        @ListOps!("indexOf", authKeys, authKey, *idxAuthKeysCh) |
+                        @ListOps!("indexOf", authPubKeys, authKey, *idxAuthPubKeysCh) |
+                        for (@idxAuthKeys <- idxAuthKeysCh; @idxAuthPubKeys <- idxAuthPubKeysCh) {
+                          if (idxAuthKeys >= 0 or idxAuthPubKeys >= 0) {
+                            for (@nonce <- @(*multiSig, *nonceStore); @nonceMap <- @(*multiSig, *nonceMapStore)) {
+                              @(*multiSig, *nonceStore)!(nonce + 1) |
+                              if (quorumSize == 1) {
+                                @(*multiSig, *nonceMapStore)!(nonceMap.set(nonce, Nil)) |
+                                new successCh in {
+                                  @revVault!("transfer", targetRevAddress, amount, revVaultAuthKey, *successCh) |
+                                  for (@(success, msg) <- successCh) {
+                                    if (success) {
+                                      ret!((true, (false, "done")))
+                                    } else {
+                                      ret!((false, msg))
+                                    }
+                                  }
+                                }
+                              } else {
+                                @(*multiSig, *nonceMapStore)!(
+                                  nonceMap.set(nonce, (targetRevAddress, amount, Set(authKey)))
+                                ) |
+                                ret!((true, (true, nonce)))
+                              }
+                            }
+                          } else {
+                            ret!((false, "invalid authKey"))
+                          }
+                        }
+                      }
+                    } |
+
+                    // Confirms transfer transaction.
+                    // returns (true, (true, nonce)) when more sigs required
+                    //         (true, (false, "done")) when only one sig required
+                    //         (false, errorMsg) on failure
+                    contract multiSig(@"confirm", @targetRevAddress, @amount, @authKey, @nonce, ret) = {
+                      // Check that authKeys is in the list of authKeys or authPubKeys
+                      new idxAuthKeysCh, idxAuthPubKeysCh in {
+                        @ListOps!("indexOf", authKeys, authKey, *idxAuthKeysCh) |
+                        @ListOps!("indexOf", authPubKeys, authKey, *idxAuthPubKeysCh) |
+                        for (@idxAuthKeys <- idxAuthKeysCh; @idxAuthPubKeys <- idxAuthPubKeysCh) {
+                          if (idxAuthKeys >= 0 or idxAuthPubKeys >= 0) {
```

Comment:
Might be worth it to create a helper method for the check here and in `multiSig!transfer`.